File: app/src/main/java/io/github/qwqgong/androidcyaml/VpnPlatformHost.java
package io.github.qwqgong.androidcyaml;

import android.app.PendingIntent;
import android.content.Context;
import android.net.VpnService;

interface VpnPlatformHost {
    Context platformContext();

    VpnService.Builder newPlatformBuilder();

    PendingIntent openAppPendingIntent();
}

File: app/src/main/java/io/github/qwqgong/androidcyaml/VpnPlatformHost.kt
package io.github.qwqgong.androidcyaml

import android.app.PendingIntent
import android.content.Context
import android.net.VpnService

interface VpnPlatformHost {
    fun platformContext(): Context

    fun newPlatformBuilder(): VpnService.Builder

    fun openAppPendingIntent(): PendingIntent
}
